src/SldStyleParser.ts

```diff
@@ -510,404 +518,463 @@ export class SldStyleParser implements StyleParser<string> {
-    const rotation: string = _get(sldSymbolizer, 'Graphic[0].Rotation[0]._');
+  getLineSymbolizerFromSldSymbolizer(sldSymbolizer: any): GsLineSymbolizer {
+    const lineSymbolizer: GsLineSymbolizer = {
+      kind: 'Line'
+    };
+    const strokeEl = get(sldSymbolizer, 'Stroke');
+    if (strokeEl.length < 1) {
+      throw new Error('LineSymbolizer cannot be parsed. No Stroke detected');
+    }
+    const color = getParameterValue(strokeEl, 'stroke', this.sldVersion);
+    const width = getParameterValue(strokeEl, 'stroke-width', this.sldVersion);
+    const opacity = getParameterValue(strokeEl, 'stroke-opacity', this.sldVersion);
+    const lineJoin = getParameterValue(strokeEl, 'stroke-linejoin', this.sldVersion);
+    const lineCap = getParameterValue(strokeEl, 'stroke-linecap', this.sldVersion);
+    const dashArray = getParameterValue(strokeEl, 'stroke-dasharray', this.sldVersion);
+    const dashOffset = getParameterValue(strokeEl, 'stroke-dashoffset', this.sldVersion);
 
-    let fillParams: any[];
-    if (this.sldVersion === '1.0.0') {
-      fillParams = _get(sldSymbolizer, 'Graphic[0].Mark[0].Fill[0].CssParameter') || [];
-    } else {
-      fillParams = _get(sldSymbolizer, 'Graphic[0].Mark[0].Fill[0].SvgParameter') || [];
+    if (color) {
+      lineSymbolizer.color = color;
     }
-    const colorIdx: number = fillParams.findIndex((cssParam: any) => {
-      return cssParam.$.name === 'fill';
-    });
-    let colorObj: any;
-    if (this.sldVersion === '1.0.0') {
-      colorObj = _get(sldSymbolizer, 'Graphic[0].Mark[0].Fill[0].CssParameter[' + colorIdx + ']');
-    } else {
-      colorObj = _get(sldSymbolizer, 'Graphic[0].Mark[0].Fill[0].SvgParameter[' + colorIdx + ']');
+    if (width !== undefined) {
+      lineSymbolizer.width = parseFloat(width);
     }
-    const color = this.getExpressionOrValueFromSldObject(colorObj);
-
-    const fillOpacityIdx: number = fillParams.findIndex((cssParam: any) => {
-      return cssParam.$.name === 'fill-opacity';
-    });
-    let fillOpacity: string = _get(sldSymbolizer,
-      'Graphic[0].Mark[0].Fill[0].CssParameter[' + fillOpacityIdx + ']._');
-    if (this.sldVersion === '1.0.0') {
-      fillOpacity = _get(sldSymbolizer,
-        'Graphic[0].Mark[0].Fill[0].CssParameter[' + fillOpacityIdx + ']._');
-    } else {
-      fillOpacity = _get(sldSymbolizer,
-        'Graphic[0].Mark[0].Fill[0].SvgParameter[' + fillOpacityIdx + ']._');
+    if (opacity !== undefined) {
+      lineSymbolizer.opacity = parseFloat(opacity);
     }
-    const markSymbolizer: MarkSymbolizer = {
-      kind: 'Mark',
-      wellKnownName: 'circle'
-    };
-
-    if (opacity) {
-      markSymbolizer.opacity = parseFloat(opacity);
+    if (lineJoin) {
+      // geostyler-style and ol use 'miter' whereas sld uses 'mitre'
+      if (lineJoin === 'mitre') {
+        lineSymbolizer.join = 'miter';
+      } else {
+        lineSymbolizer.join = lineJoin as 'bevel' | 'miter' | 'round' | undefined;
```

Comment:
Maybe it's time to create a dedicated type for linejoin and linecap in geostyler-style.

Reply:
https://github.com/geostyler/geostyler-style/issues/499